### PR TITLE
Refactors smoke tests to use testenvironment as a param

### PR DESCRIPTION
**Pull request by kalverra**
Also moves `CreateNodeKeysBundle` out of the testing framework, and fixes incompatibility issues that slipped through on last release.

### CODE REVIEW

**Review by AnieeG — APPROVED** [review]

File: integration-tests/contracts/ethereum_contracts.go [hunk 1]
```diff
@@ -444,7 +441,7 @@ func NewVRFConsumerRoundConfirmer(
 }
 
 // ReceiveBlock will query the latest VRFConsumer round and check to see whether the round has confirmed
-func (f *VRFConsumerRoundConfirmer) ReceiveBlock(block blockchain.NodeBlock) error {
+func (f *VRFConsumerRoundConfirmer) ReceiveBlock(header blockchain.NodeHeader) error {
```
Comment (AnieeG) [inline integration-tests/contracts/ethereum_contracts.go:444]:
```suggestion
func (f *VRFConsumerRoundConfirmer) ReceiveHeader(header blockchain.NodeHeader) error {
```

File: integration-tests/contracts/ethereum_contracts.go [hunk 2]
```diff
@@ -830,7 +827,7 @@ func NewRunlogRoundConfirmer(
 }
 
 // ReceiveBlock will query the latest Runlog round and check to see whether the round has confirmed
-func (o *RunlogRoundConfirmer) ReceiveBlock(_ blockchain.NodeBlock) error {
+func (o *RunlogRoundConfirmer) ReceiveBlock(_ blockchain.NodeHeader) error {
```
Comment (AnieeG) [inline integration-tests/contracts/ethereum_contracts.go:830]:
```suggestion
func (o *RunlogRoundConfirmer) ReceiveHeader(_ blockchain.NodeHeader) error {
```

File: integration-tests/contracts/ethereum_contracts.go [hunk 3]
```diff
@@ -456,7 +453,7 @@ func (f *VRFConsumerRoundConfirmer) ReceiveBlock(block blockchain.NodeBlock) err
 		Str("Contract Address", f.consumer.Address()).
 		Int64("Waiting for Round", f.roundID.Int64()).
 		Int64("Current round ID", roundID.Int64()).
-		Uint64("Block Number", block.NumberU64())
+		Uint64("Block Number", header.Number.Uint64())
```
Comment (AnieeG) [inline integration-tests/contracts/ethereum_contracts.go:456]:
```suggestion
		Uint64("Header Number", header.Number.Uint64())
```

File: integration-tests/contracts/ethereum_keeper_contracts.go [hunk 4]
```diff
@@ -592,7 +592,7 @@ func NewKeeperConsumerRoundConfirmer(
 }
 
 // ReceiveBlock will query the latest Keeper round and check to see whether the round has confirmed
-func (o *KeeperConsumerRoundConfirmer) ReceiveBlock(_ blockchain.NodeBlock) error {
+func (o *KeeperConsumerRoundConfirmer) ReceiveBlock(_ blockchain.NodeHeader) error {
```
Comment (AnieeG) [inline integration-tests/contracts/ethereum_keeper_contracts.go:595]:
```suggestion
func (o *KeeperConsumerRoundConfirmer) ReceiveHeader(_ blockchain.NodeHeader) error {
```

**Review by AnieeG — DISMISSED** [review]

**Review by tateexon — APPROVED** [review]

File: integration-tests/contracts/ethereum_keeper_contracts.go [hunk 5]
```diff
@@ -592,7 +592,7 @@ func NewKeeperConsumerRoundConfirmer(
 }
 
 // ReceiveBlock will query the latest Keeper round and check to see whether the round has confirmed
```
Comment (gheorghestrimtu) [inline integration-tests/contracts/ethereum_keeper_contracts.go:594]: Refactor comments also

File: integration-tests/smoke/cron_test.go [hunk 6]
```diff
@@ -25,12 +25,86 @@ import (
...
+				environment.New(&environment.Config{}).
+					AddHelm(mockservercfg.New(nil)).
+					AddHelm(mockserver.New(nil)).
+					AddHelm(ethereum.New(nil)).
+					AddHelm(chainlink.New(0, map[string]interface{}{
+						"env": networks.SimulatedEVM.ChainlinkValuesMap(),
+					})),
+			),
+			Entry("Cronjob suite on General EVM Network read from env vars @general",
+				networks.GeneralEVM(),
```
Comment (gheorghestrimtu) [inline integration-tests/smoke/cron_test.go:39]: we could use something like this to reduce code size
```
func DefaultEnvForNetwork(client *blockchain.EVMNetwork) *e.Environment {
	return environment.New(&environment.Config{}).
		AddHelm(mockservercfg.New(nil)).
		AddHelm(mockserver.New(nil)).
		AddHelm(ethereum.New(&ethereum.Props{
			NetworkName: client.Name,
			Simulated:   client.Simulated,
			WsURLs:      client.URLs,
		})).
		AddHelm(chainlink.New(0, map[string]interface{}{
			"env": client.ChainlinkValuesMap(),
		}))
}
```

File: integration-tests/smoke/cron_test.go [hunk 7]
```diff
@@ -25,12 +25,86 @@ import (
...
+						"env": networks.MetisStardust.ChainlinkValuesMap(),
+					})),
+			),
+			Entry("Cronjob suite on Sepolia Testnet @sepolia",
+				networks.SepoliaTestnet,
+				environment.New(&environment.Config{}).
+					AddHelm(mockservercfg.New(nil)).
+					AddHelm(mockserver.New(nil)).
+					AddHelm(ethereum.New(&ethereum.Props{
+						NetworkName: networks.MetisStardust.Name,
```
Comment (skudasov) [inline integration-tests/smoke/cron_test.go:72]: Sepolia URL needed?
In fact, is it required for cron job?

File: integration-tests/smoke/ocr_test.go [hunk 8]
```diff
@@ -55,26 +141,12 @@ var _ = Describe("OCR Feed @ocr", func() {
...
-		}).
-			AddHelm(mockservercfg.New(nil)).
-			AddHelm(mockserver.New(nil)).
-			AddHelm(evmChart).
-			AddHelm(chainlink.New(0, map[string]interface{}{
-				"env":      testNetwork.ChainlinkValuesMap(),
-				"replicas": 6,
-			}))
+		testEnvironment = env
+		testEnvironment.Cfg.NamespacePrefix = fmt.Sprintf("smoke-ocr-%s", strings.ReplaceAll(strings.ToLower(testNetwork.Name), " ", "-"))
```
Comment (skudasov) [inline integration-tests/smoke/ocr_test.go:148]: Can we rename them properly not in test and runtime?

**Review by skudasov — DISMISSED** [review]

**Review by skudasov — DISMISSED** [review]

**Review by AnieeG — DISMISSED** [review]

**Review by skudasov — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**